pkg/trace/api/telemetry.go

```diff
@@ -44,29 +45,29 @@ const (
 	cloudResourceIdentifierHeader string            = "dd-cloud-resource-identifier"
 )
 
-// telemetryMultiTransport sends HTTP requests to multiple targets using an
-// underlying http.RoundTripper. API keys are set separately for each target.
-// The target hostname
-// When multiple endpoints are in use the response from the main endpoint
-// is proxied back to the client, while for all aditional endpoints the
-// response is discarded. There is no de-duplication done between endpoint
-// hosts or api keys.
+// TelemetryForwarder sends HTTP requests to multiple targets.
+// The handler returns immediately and the forwarding is done in the background.
 //
-// Could be extended in the future to allow supporting more product endpoints
-// by simply parametrizing metric tags, and logger names
-type telemetryMultiTransport struct {
-	Transport http.RoundTripper
-	Endpoints []*config.Endpoint
-	statsd    statsd.ClientInterface
+// To provide somne backpressure, we limit the number of concurrent forwarded requests
+type TelemetryForwarder struct {
+	endpoints []*config.Endpoint
+	conf      *config.AgentConfig
+
+	inflightWaiter sync.WaitGroup
+	inflightCount  atomic.Int32
+	maxConcurrent  int32
```

Comment:
@raphaelgavache raises a good point that this should be capped by the amount of memory requests are using rather than their numbers

Reply:
Fixed